File: Proiect/src/main/java/repository/RepoDBOrganiser.java
package repository;

import model.JdbcUtils;
import model.Organiser;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Properties;

public class RepoDBOrganiser implements Repository<Organiser, Integer> {

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public RepoDBOrganiser(Properties props) {
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public void add(Organiser el) {
        logger.traceEntry("saving organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into Organiser (Id, Name) values (?, ?)")){

            preparedStatement.setInt(1, el.getID());
            preparedStatement.setString(2, el.getName());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Organiser el) {
        logger.traceEntry("deleting organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("delete from Organiser where Id = ?")){

            preparedStatement.setInt(1, el.getID());
            int result = preparedStatement.executeUpdate();
            logger.trace("Removed {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Organiser el, Integer id) {
        logger.traceEntry("updating organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("Update Organiser set Name = ? where Id = ?")){

            preparedStatement.setString(1, el.getName());
            preparedStatement.setInt(2, el.getID());
            int result = preparedStatement.executeUpdate();
            logger.trace("Updated {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public Organiser findById(Integer id) {
        logger.traceEntry("Finding organiser {}", id);
        Connection con = dbUtils.getConnection();
        Organiser organiser = new Organiser();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser where Id = ?")){
            preparedStatement.setInt(1, id);
            try(ResultSet result = preparedStatement.executeQuery()) {

                while(result.next()){
                    int Id = result.getInt("Id");
                    String name = result.getString("Name");
                    organiser.setId(Id);
                    organiser.setName(name);
                logger.trace("Found {} instances", organiser);
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return organiser;
    }

    @Override
    public Iterable<Organiser> findAll() {
        logger.traceEntry("find all organisers");
        Connection con = dbUtils.getConnection();
        List<Organiser> organisers = new ArrayList<>();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("Id");
                    String name = result.getString("Name");
                    Organiser organiser = new Organiser(id,name);
                    organisers.add(organiser);
                }
            }
        }
        catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB" + e);
        }
        logger.traceExit(organisers);
        return organisers;
    }

    @Override
    public Collection<Organiser> getAll() {
        logger.traceEntry("get all organisers");
        Connection con = dbUtils.getConnection();
        List<Organiser> organisers = new ArrayList<>();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("Id");
                    String name = result.getString("Name");
                    Organiser organiser = new Organiser(id,name);
                    organisers.add(organiser);
                }
            }
        }
        catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB" + e);
        }
        logger.traceExit(organisers);
        return organisers;
    }
}

File: Csharp/MPP/MPP/repository/RepoDBOrganiser.cs
using System.Data;
using MPP.model;
using System;
using log4net;
using System.Collections.Generic;

namespace MPP.repository
{
    public class RepoDBOrganiser : IRepoOrganiser
    {

        //private static readonly ILog log = LogManager.GetLogger("SortingTaskDbRepository");
        private static readonly ILog log = LogManager.GetLogger("RepoDBOrganiser");


        IDictionary<String, string> props;

        public RepoDBOrganiser(IDictionary<String, string> props)
        {
            log.Info("Creating RepoOrganiserDB");
            this.props = props;
        }

        public void add(Organiser entity)
        {
            log.InfoFormat("saving organiser {0}", entity);
            var con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into Organiser  values (@Id, @Name,@Password)";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@Id";
                paramId.Value = entity.Id;
                comm.Parameters.Add(paramId);

                var paramName = comm.CreateParameter();
                paramName.ParameterName = "@Name";
                paramName.Value = entity.Name;
                comm.Parameters.Add(paramName);

                var paramPass = comm.CreateParameter();
                paramPass.ParameterName = "@Password";
                paramPass.Value = entity.Password;
                comm.Parameters.Add(paramPass);

                var result = comm.ExecuteNonQuery();
                log.InfoFormat("Saved {0} instance", result);
                if (result == 0)
                {
                    log.Error("No organiser added !");
                    throw new Exception("No organiser added !");
                }
            }

        }

        public void delete(Organiser entity)
        {
            log.InfoFormat("deleting organiser {0}", entity);
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Organiser where Id=@Id";

                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@Id";
                paramId.Value = entity.Id;
                comm.Parameters.Add(paramId);

                var dataR = comm.ExecuteNonQuery();
                log.InfoFormat("Deleted {0} instance", dataR);
                if (dataR == 0)
                {
                    log.Error("No organiser deleted!");
                    throw new Exception("No organiser deleted!");
                }
            }
        }

        public void update(Organiser entity, int id)
        {
            log.InfoFormat("updating organiser {0} with {1}", id, entity);
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "update Organiser set Name=@Name  where Id=@Id";

                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@Id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                var paramName = comm.CreateParameter();
                paramName.ParameterName = "@Name";
                paramName.Value = entity.Name;
                comm.Parameters.Add(paramName);

                var dataR = comm.ExecuteNonQuery();
                log.InfoFormat("Updated {0} instance", dataR);
                if (dataR == 0)
                {
                    log.Error("No organiser  updated!");
                    throw new Exception("No organiser updated!");
                }
            }
        }

        public Organiser findbyId(int id)
        {
            log.InfoFormat("finding one organiser {0}", id);
            var con = DBUtils.getConnection(props);
            Organiser organiser = new Organiser();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select Id,Name,Password from Organiser where Id=@Id";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@Id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idb = dataR.GetInt32(0);
                        string name = dataR.GetString(1);
                        string pass = dataR.GetString(2);
                        organiser.Id = idb;
                        organiser.Name = name;
                        organiser.Password = pass;
                    }
                }
            }
            log.InfoFormat("Found {0} instance", organiser);
            return organiser;
        }

        public ICollection<Organiser> findAll()
        {
            log.InfoFormat("finding all organisers");
            IDbConnection con = DBUtils.getConnection(props);
            IList<Organiser> organisers = new List<Organiser>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select Id,Name,Password from Organiser";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        String name = dataR.GetString(1);
                        String pass = dataR.GetString(2);
                        Organiser organiser = new Organiser(id, name, pass);
                        organisers.Add(organiser);
                    }
                }
            }
            log.InfoFormat("found {0} organisers", organisers.Count);
            return organisers;
        }

        public Organiser findByNameAndPassword(string name, string password)
        {
            log.InfoFormat("finding organiser by name {0}", name, password);
            var con = DBUtils.getConnection(props);
            Organiser organiser = new Organiser();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select Id,Name,Password from Organiser where Name=@Name and Password=@Password";
                var paramName = comm.CreateParameter();
                paramName.ParameterName = "@Name";
                paramName.Value = name;
                comm.Parameters.Add(paramName);

                var paramPass = comm.CreateParameter();
                paramPass.ParameterName = "@Password";
                paramPass.Value = password;
                comm.Parameters.Add(paramPass);

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idb = dataR.GetInt32(0);
                        string orgName = dataR.GetString(1);
                        string pass = dataR.GetString(2);
                        organiser.Id = idb;
                        organiser.Name = orgName;
                        organiser.Password = pass;
                    }
                }
            }
            log.InfoFormat("Found {0} instance", organiser);
            return organiser;
        }
    }
}

